src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/SerializationRecordId.cs

```diff
@@ -45,14 +44,5 @@ internal static SerializationRecordId Decode(BinaryReader reader)
     public override bool Equals(object? obj) => obj is SerializationRecordId other && Equals(other);
 
     /// <inheritdoc />
-    public override int GetHashCode()
-    {
-        int id = _id;
-#if NET
-        Span<int> integers = new(ref id);
-#else
-        Span<int> integers = stackalloc int[1] { id };
-#endif
-        return (int)XxHash32.HashToUInt32(MemoryMarshal.AsBytes(integers));
-    }
+    public override int GetHashCode() => _id;
```

Comment:
This is done on purpose, to prevent from hashcode collision attacks (the attacker can create a payload with such Ids that would cause OOM in the dictionary)

Reply:
There's no randomization in xxhash32. this is just a deterministic mapping from one integer to another, with the only possibility of creating more collusions. to mitigate study attacks, this should use a random seed.

Reply:
> There's no randomization in xxhash32. this is just a deterministic mapping from one integer to another, with the only possibility of creating more collisions. to mitigate such attacks, this should use a random seed, eg System.HashCode

Thanks, I was not aware of that. I've assumed that it's the right thing to do since it got approved by @bartonjs.

This library needs to target Full Framework and .NET Standard, so we can not use `System.HashCode` for all TFMs. But... w can use it for `#NET`. Would that solve the problem for windowsdesktop repo?

Reply:
> so we can not use System.HashCode for all TFMs

We can:
https://www.nuget.org/packages/Microsoft.Bcl.HashCode/

But if the goal here is just to include some randomness with a single integer, I'd assume we can just do something like:
```C#
private static readonly int s_seed = new Random().Next(-int.MaxValue, int.MaxValue);

public int GetHashCode() => _id ^ s_seed;
```

Reply:
> We can: https://www.nuget.org/packages/Microsoft.Bcl.HashCode/

Nice, I was not aware of the fact that such package exists. It's already used by Cbor, so we should be fine using it?

https://github.com/dotnet/runtime/blob/9c51d32db656a832cdb23769ec7c63b1dd1d9e34/src/libraries/System.Formats.Cbor/src/System.Formats.Cbor.csproj#L56

> But if the goal here is just to include some randomness with a single integer
> or some other form of incorporation that Jeremy and Levi bless.

I would prefer to wait for @bartonjs or @GrabYourPitchforks  feedback.

@stephentoub would you mind updating your PR to switch to `HashCode` type or would you prefer me to do it?